### ISSUE TITLE
`maxCoresPercentageToUseForBackgroundIndexing` is too high by default

### ISSUE THREAD

**Issue by MahdiBM** [swiftlang/sourcekit-lsp#1836]
### Swift version

Swift 6.0.2 release

### Platform

-

### Editor

VS Code

### Description

`maxCoresPercentageToUseForBackgroundIndexing` is set to 100% by default. I think that's too much.
I'd say let's lower it to 80/70% and anyone who wants maximum performance can increase it.
It's quite rare that something makes my M1 Macbook's fans spin ... and background indexing did achieve that.

### Steps to Reproduce

Enabled background indexing on a big enough project.

**Comment by ahoppen** [comment]
Synced to Apple’s issue tracker as rdar://140186846

**Comment by ahoppen** [comment]
`maxCoresPercentageToUseForBackgroundIndexing` is a little misleading and maybe we should remove the setting altogether. Even when setting it to 0.8, it might use all your CPU cores because background preparation invokes `swift build --experimental-prepare-for-indexing` and we don’t know how many cores that takes and we assume that it’s a single one. 

All background activity happens with a lower priority on the OS level (high `nice` value, set [here](https://github.com/swiftlang/sourcekit-lsp/blob/a8643915ab75ade2ac8ca69c2938f4e4e99fb713/Sources/TSCExtensions/Process%2BRun.swift#L177)) unless it’s required for functionality in the current file. This should ensure that other higher priority tasks take precedence. And I think using system functionality like this is the better than trying to do process scheduling ourselves by limiting the number of cores we use.

**Comment by ahoppen** [comment]
Closing this because `maxCoresPercentageToUseForBackgroundIndexing` has been hidden from the configuration options by https://github.com/swiftlang/sourcekit-lsp/pull/1984 and artificially reducing the number of CPU cores used for indexing should not be the solution as I described in my last comment.

**Comment by MahdiBM** [comment]
@ahoppen I agree tbh, but artificially reducing the cpu usage might be better than nothing.
Just today I heard from someone that their machine was struggling and lagging due to the high CPU usage. I prompted them to file an issue but not sure if they will file the issue. Just saying that there is an issue to be resolved. I personally have gotten used to it + It doesn't make my M1 macbook _lag_.
